gammapy/spectrum/fit.py

```diff
@@ -204,8 +204,11 @@ def _predict_counts_helper(self, obs, model, forward_folded=True):
                                               e_reco=binning)
             counts = temp.data.data
         else:
-            # TODO: This could also be part of calculate predicted counts
-            counts = model.integral(binning[:-1], binning[1:])
+            temp = calculate_predicted_counts(model=model,
```

Comment:
I assumed that `model.integral(binning[:-1], binning[1:])` always returns a flux. If it returns counts the code is probably fine. The documentation should be improved in this case I guess.